node/src/blockchain/blockchain_interface/mod.rs

```diff
@@ -1,52 +1,39 @@
 // Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
-pub mod blockchain_interface_null;
 pub mod blockchain_interface_web3;
 pub mod data_structures;
 pub mod lower_level_interface;
 pub mod test_utils;
 
-use crate::accountant::db_access_objects::payable_dao::PayableAccount;
 use crate::accountant::scanners::mid_scan_msg_handling::payable_scanner::blockchain_agent::BlockchainAgent;
-use crate::blockchain::blockchain_bridge::PendingPayableFingerprintSeeds;
 use crate::blockchain::blockchain_interface::data_structures::errors::{
-    BlockchainAgentBuildError, BlockchainError, PayableTransactionError, ResultForReceipt,
-};
-use crate::blockchain::blockchain_interface::data_structures::{
-    ProcessedPayableFallible, RetrievedBlockchainTransactions,
+    BlockchainAgentBuildError, BlockchainError,
 };
+use crate::blockchain::blockchain_interface::data_structures::RetrievedBlockchainTransactions;
 use crate::blockchain::blockchain_interface::lower_level_interface::LowBlockchainInt;
-use crate::db_config::persistent_configuration::PersistentConfiguration;
 use crate::sub_lib::wallet::Wallet;
-use actix::Recipient;
-use web3::types::{Address, BlockNumber, H256};
+use futures::Future;
+use masq_lib::blockchains::chains::Chain;
+use web3::types::{Address, BlockNumber};
 
 pub trait BlockchainInterface {
     fn contract_address(&self) -> Address;
 
+    fn get_chain(&self) -> Chain;
+
+    fn lower_interface(&self) -> Box<dyn LowBlockchainInt>;
```

Comment:
Redesign issue solution.
There is only one function in the trait that requires the transport `get_contract`.
Once we implement associated type defaults on the trait LowBlockchainInt it will eliminate this dependency on web3 such that this trait could be implement on any blockchain.

Using as a reference.
We intentionally created the structor  without a reference to prevent lifetime issues, the idea is to create it where ever necessary and then delete it.

If these points don't sound reasonable then we could look at a redesign.

Reply:
First, I've heard many times from you two about so called "associated type defaults". I've got an issue resolving what it is supposed to mean. In my perspective, even the obsolete Rust version embraces features that would allow you what I think you wish to achieve. Traits can easily be declared together with an arbitrary number of associated types. The word "default" is the confusing part. I agree with you that the associated types cannot be defined with a zero configuration, the defaulted type. However, I'm asking why do you need to have this ability? I think it is not a requirement and so the constraints you talk about are wrongly put in consideration, they don't exist and should not hinder you at all. 

Second, I talked about it in the initial comment that it's quite uncertain that we will always be able to create the returned structure "in place". Maybe we will be unable to clone or we would have to store some inputs needed to construct the structure inside the higher level interface for the entire time the program is running. It could at least be awkward. 
So, I'm only suggesting a solution which will be effective (a reference) and will be always supported (at least it's much harder to hit some constrains there).
Utkarsh's deepest will to eliminate every possible lifetime no matter what sometimes gives me shivers. Rust is quite oriented this direction, it's an absolutely normal practise to use references extensively. It's true that lifetimes may look intimidating but let's not treat them as "Rust cancer". They're useful and important part of the language. This defensive effort leads to overly used cloning etc. If anything, it's not of concern, that this is presented to Syther as if it should be a mantra. If you are blindly motivated to eliminate every lifetime, you become less educated on lifetimes because of the lack of practise in using them. Of course, it also makes you think that "they are tough". But they shouldn't be. If you think so, the issue is that you are too intimidated with them. 
Lastly, by lacking this knowledge, you deliberately disqualify you from solving certain design situations that could have a solution, but only with their usage. 
I always say "go and look at some other Rust open-source projects", or simply click through some of our third party libraries, you will see plenty of lifetimes everywhere. Because they are so natural to Rust, it's deep in its DNA.

**STATUS: I'D BE HAPPY ENOUGH IF YOU PROVIDED A COMMENT TO THE TRAIT DECLARATION REFERRING ABOUT THIS POSSIBLE ISSUE.** I don't think it should be forgotten and if you do nothing it will definitely escape your minds. Please take time to write a decent abstract of my concern for those who could bump into it in the future. Except the comment, you can leave it as it is.

Reply:
We have added a comment as requested :)